index.bs

```diff
@@ -426,19 +426,25 @@ particularly important for the data structure described in [[#queue-with-sizes]]
  const reader = readableStream.getReader({ mode: "byob" });
 
  let startingAB = new ArrayBuffer(1024);
- readInto(startingAB)
-   .then(buffer => console.log("The first 1024 bytes:", buffer))
-   .catch(e => console.error("Something went wrong!", e));
-
- function readInto(buffer, offset = 0) {
-   if (offset === buffer.byteLength) {
-     return Promise.resolve(buffer);
-   }
+ const buffer = await readInto(startingAB);
+ console.log("The first 1024 bytes: ", buffer);
+
+ async function readInto(buffer) {
+   let offset = 0;
+   let view;
+   let done;
+
+   do {
+     ({value: view, done} =
```

Comment:
I think this would read better as:
```javascript
const {value: view, done} = await reader.read(new Uint8Array(buffer, offset, buffer.byteLength - offset));
```
Then we don't need the extra parentheses or the `let view` and `let done` declarations at the top.

Reply:
Good! I hadn't noticed that we were no longer using `view` and `done` outside the loop.

Reply:
That makes sense, I've just added your suggestion in my last commit. Thank you for taking a look and the suggestions as well! 😃